pkg/server/ratelimited.go

```diff
@@ -0,0 +1,62 @@
+package server
+
+import (
+	"fmt"
+	"net/http"
+	"sync"
+	"time"
+
+	"golang.org/x/time/rate"
+)
+
+// ErrWriteLimitReached is an error that is returned when a cluster has sent too many requests.
+type ErrWriteLimitReached string
+
+func (e ErrWriteLimitReached) Error() string {
+	return fmt.Sprintf("write limit reached for key %q", string(e))
```

Comment:
maybe we should rename `key` to `id` or `cluster` now

Reply:
`write limit reached for cluster with id`? :blush: